types/ember__debug/index.d.ts

```diff
@@ -87,5 +87,23 @@ export function deprecate(
          * An optional url to the transition guide on the emberjs.com website.
          */
         url?: string;
+        /**
+         * The library emitting this deprecation
+         */
+        for: string;
+        /**
+         * Information about the stage for this deprecation
+         */
+        since: {
```

Comment:
The RFC text wasn't especially clear about this, so I'm actually going to submit a PR to amend it and clarify this, but adding these keys as *required* would actually be a breaking change. These are *currently* optional keys, but calling `deprecate` without them is deprecated (see [the implementation][impl]).

For the sake of better user feedback, let's actually tackle this via an overload:

- extract the options into a standalone interface

- on the base version of the interface, make the `for` and `since` keys optional

- extend the base/legacy interface with an interface where those keys are required

- create two declarations for `deprecate`:

    - one with the non-deprecated options

    - one with the deprecated (looser) options, marked with `@deprecate` in the JSDoc docstring

    (They must go in this order for the deprecation handling in the TS language server to work correctly.)

This will let users who have the types installed (including via automatic type acquisition in VS Code) see that the legacy form is deprecated. You can see this working in [this playground][play], and here's how it looks for end users:

<img width="679" alt="Screen Shot 2021-04-16 at 08 03 42" src="https://user-images.githubusercontent.com/2403023/115035914-4548b380-9e8a-11eb-9d3e-a85d3af38cde.png">

[impl]: https://github.com/emberjs/ember.js/pull/19133/files#diff-f88aff7509fd7b43092f0a6d015f533761c7a9b2526f00495b1df05951564a3eR208-R232
[play]: https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCIQA5QguZgD2IA8hvkQM7IDeAUMssACYBcyFYUoA5gNz3IAruGAAbNhy4g+AwVFEB+CZx78GMAlCXsV0te1BJtdBgzgA3OGLgAjURGVSZp5BBC37rHU-0BfWv60oJCwiCjoWDh4hCAAymBw3Dyk5CBUEAAekCBMVBHYuKkpMVQmphpQjqoCDBSGDjQ1LsgWVqIeDZLVzaZuHV5dek3+gQD0AFTj9ONomAV4KHDIYAAWPAB006O0TDjt2MgwwgipyLuRhRAAFARkJWz5Uanxicl3lACUbOYEzPy0EymyBmj0uLWWa2km2ByAAAud5pAmAYQEhkABmdYAJgALOsAAzIACeBEEyAAtoIOMhcKJRMgAAYIp4QBnIADuwFWEIgWx2ezgByOqNOzMuN3eaQecyeMWKn2+vyY-zFCyuZQYzDYAHIMAQEmACNqADRNYT4cTIbUVCDmaAms3yHUrMBgDAUFijUaZODkjD2dYIAjkh2+D4qmXijWMLy6-V4I2mlzmsQ6m12qAO5NOq0ut0er0+v0BoMhpPlTQ6kAQPAQngULOmOqohrRlytax2BrakAEdmNnp9Lux4Dk8kQJjABaiIkD5AjcO0IA

Reply:
Just FYI - I merged the verbiage tweak in https://github.com/emberjs/rfcs/pull/736 (making the keys optional for 3.x).